### PR TITLE
add CAN model

### PR DESCRIPTION
**Pull request by zhangjunlongtech**
See README for details on how the model works

The accuracy of the current given model is 52.8%, and the verification information is as follows
CAN_eval

The accuracy is higher than that of similar benchmarking models (51.72%).

ps: please don't squash commits

### CODE REVIEW

File: research/huawei-ostd/CAN/can/models/builder.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
+"""
+build models
```
Comment (wxsIcey) [inline research/huawei-ostd/CAN/can/models/builder.py:2]: 每个文件开头应保持一致

File: research/huawei-ostd/CAN/can/data/base_dataset.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+    Args:
+        - data_dir:
+        - label_file:
+        - output_columns (List(str)): names of elements in the output tuple of __getitem__
+    Attributes:
+        data_list (List(Tuple)): source data items (e.g., containing image path and raw annotation)
+    """
+
+    def __init__(
```
Comment (wxsIcey) [inline research/huawei-ostd/CAN/can/data/base_dataset.py:19]: 函数参数有的注明类型，有的没有，一个项目中的文件应该一致

File: research/huawei-ostd/CAN/tests/test_can_model.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
+import sys
+import math
+import mindspore as ms
+
+from can.models.backbones import build_backbone
+from can.models.heads import build_head
+from can.models import build_model
+from mindspore import ops
+
+sys.path.append(".")
```
Comment (wxsIcey) [inline research/huawei-ostd/CAN/tests/test_can_model.py:10]: 这两行内容可以不用写

File: research/huawei-ostd/CAN/tests/test_can_preprocess.py [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+        print(data["label_len"])
+
+
+if __name__ == "__main__":
+    """
+        Select the test function as needed
+    """
+
+    test_transform_pipeline()
+    # test_train_data_config()
```
Comment (wxsIcey) [inline research/huawei-ostd/CAN/tests/test_can_preprocess.py:226]: 不应该出现对代码的注释

File: research/huawei-ostd/CAN/tools/predict_can.py [hunk 5]
```diff
@@ -0,0 +1,318 @@
+"""
+Text recognition inference
+"""
+import logging
+import os
+import sys
+from time import time
+
+import numpy as np
+from config import parse_args
```
Comment (wxsIcey) [inline research/huawei-ostd/CAN/tools/predict_can.py:10]: 从项目中引入的函数应放在从mindspore引入算子之后

File: research/huawei-ostd/CAN/can/optim/optim_factory.py [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+    if weight_decay and filter_bias_and_bn:
+        if not isinstance(params[0], dict):  # check whether param grouping strategy is encoded in `params`
+            params = init_group_params(params, weight_decay)
+        else:
+            _logger.warning(
+                "Customized param grouping strategy detected in `params`. "
+                "filter_bias_and_bn (default=True) will be disabled"
+            )
+
+    # opt_args = dict(**kwargs)
```
Comment (wxsIcey) [inline research/huawei-ostd/CAN/can/optim/optim_factory.py:78]: 在项目中不应该出现代码注释

File: research/huawei-ostd/CAN/tools/predict_can.py (hunk 5 above)
Comment (zhangjunlongtech) [reply research/huawei-ostd/CAN/tools/predict_can.py:14]: 已修改

File: research/huawei-ostd/CAN/tests/test_can_preprocess.py (hunk 4 above)
Comment (zhangjunlongtech) [reply research/huawei-ostd/CAN/tests/test_can_preprocess.py]: 已修改

File: research/huawei-ostd/CAN/can/optim/optim_factory.py (hunk 6 above)
Comment (zhangjunlongtech) [reply research/huawei-ostd/CAN/can/optim/optim_factory.py]: 已修改

**Review by CaitinZhao — APPROVED** [review]

**Pull request merged**